compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2157,9 +2163,22 @@ static LogicalResult setConvRootConfig(mlir::FunctionOpInterface entryPointFn,
     scalableTileFlags.emplace_back(numTilingDims, false);
   }
 
+  DictionaryAttr pipelineConfig;
+  if (vecPreProcStrategy == VectorPreProcStrategy::Peeling) {
+    // Enable peeling. To this end, attach extra info to the pipeline config.
+    // This will later be extracted by LLVMCPULowerExecutableTargetPass.
+    auto ctx = convOp.getContext();
+    SmallVector<NamedAttribute> attrs;
+    auto peelAttrName = getPeelAttrName(ctx);
+    auto peelAttrVal = BoolAttr::get(ctx, true);
+    attrs.emplace_back(peelAttrName, peelAttrVal);
+    pipelineConfig = DictionaryAttr::get(ctx, attrs);
```

Comment:
nit: I think this could be clearer with some naming tweaks + a few less steps:
```c++
    auto context = convOp.getContext();
    auto enableLoopPeelingAttrName = getEnableLoopPeelingAttrName(context);
    auto trueValue = BoolAttr::get(context, true);
    pipelineConfig = DictionaryAttr::get(
        context,
        ArrayRef<NamedAttribute>({enableLoopPeelingAttrName, trueValue}));
```